src/syft/core/pointer/pointer.py

```diff
@@ -587,7 +594,7 @@ def __del__(self) -> None:
         if self.gc_enabled:
             self.client.gc.apply(self)
 
-    def _resolve_pointer_type(self):
+    def _resolve_pointer_type(self: AbstractPointer) -> AbstractPointer:
```

Comment:
Hello @tudorcebere!
What should I type instead of `(self: AbstractPointer)` to avoid mypy error?